src/lib/nodes/rate-info-service/RateInfoClient.ts

```diff
@@ -0,0 +1,33 @@
+import { Client } from '@/lib/nodes/abstract.client.ts'
+import { RateInfoResponse, RateInfoService } from '@/lib/nodes/rate-info-service/RateInfoService.ts'
+
+export class RateInfoClient extends Client<RateInfoService> {
+  constructor(endpoints: string[] = [], minNodeVersion = '0.0.0') {
+    super('adm', 'service')
+    this.nodes = endpoints.map((endpoint) => new RateInfoService(endpoint))
+    this.minNodeVersion = minNodeVersion
+
+    void this.watchNodeStatusChange()
+  }
+
+  async getAllRates(): Promise<RateInfoResponse> {
+    const node = await this.fetchNode()
```

Comment:
I think you can reuse `getNode()` from parent class and remove `async fetchNode()` below.

```suggestion
    const node = this.getNode()
```